### PR TITLE
Semeval 2012

### PR DESCRIPTION
**Pull request by jlowryduda**

### CODE REVIEW

File: conceptnet5/vectors/evaluation/analogy.py [hunk 1]
```diff
@@ -165,8 +221,144 @@ def eval_analogies(frame):
...
+
+    # Compute Spearman correlation of our ranks and MT ranks
+    our_semeval_scores = [score for pair, score in sorted(our_pair_scores.items())]
+    turk_semeval_scores = [score for pair, score in turk_rank]
+    spearman = round(spearmanr(our_semeval_scores, turk_semeval_scores)[0], 3)
+    spearman_results = confidence_interval(spearman, total)
+
+    # Compute an accuracy score on MaxDiff questions
+    maxdiff = round((correct_least + correct_most) / (total + total), 3)
+    low_maxdiff, high_maxdiff = proportion_confint((correct_least + correct_most), (2 * total))
```
Comment (rspeer) [inline conceptnet5/vectors/evaluation/analogy.py]: Why is this `total + total` in one line, but `2 * total` in the next?

File: conceptnet5/vectors/evaluation/analogy.py [hunk 2]
```diff
@@ -165,8 +221,144 @@ def eval_analogies(frame):
...
+    """
+    vectors = VectorSpaceWrapper(frame=frame)
+    results = empty_comparison_table()
+
+    if tune_analogies:
+        sat_weights = tune_pairwise_analogies(eval_pairwise_analogies, vectors, analogy_filename)
+        semeval_weights = tune_pairwise_analogies(eval_semeval2012_global, vectors)
+    else:
+        sat_weights = (0.35, 0.65)
+        semeval_weights = (0.2, 1.0)
```
Comment (rspeer) [inline conceptnet5/vectors/evaluation/analogy.py]: It looks like these came from the two ends of the scale that's currently in `tune_pairwise_analogies`, right? I'm trying it with a broader range of values.

File: conceptnet5/vectors/evaluation/analogy.py (hunk 1 above)
Comment (jlowryduda) [inline conceptnet5/vectors/evaluation/analogy.py]: No reason other than sloppiness.

File: conceptnet5/vectors/evaluation/analogy.py (hunk 2 above)
Comment (jlowryduda) [inline conceptnet5/vectors/evaluation/analogy.py]: Exactly. Trying a broader range is a good idea.

Comment (rspeer) [inline conceptnet5/vectors/evaluation/analogy.py]: I set the dataset to 'dev' and tried a broader range. `(0.05, 1.5)` came out as better parameters.

I then evaluated with these parameters on the 'test' dataset and got an identical result. Is this a coincidence, a mixup in my commands, or am I getting the same data each time?

The first command I ran was:

```
evaluate(vsw.frame, 'data/raw/analogy/SAT-package-V3.txt', subset='dev', tune_analogies=True)
```

And the second was:

```
evaluate(vsw.frame, 'data/raw/analogy/SAT-package-V3.txt', subset='test', tune_analogies=False)
```

Comment (rspeer) [inline conceptnet5/vectors/evaluation/analogy.py]: To clarify, before running the second command, I edited the default parameters to be (0.05, 1.5).

Comment (rspeer) [inline conceptnet5/vectors/evaluation/analogy.py]: Okay, I just messed something up before, this works.

**Comment by rspeer** [comment]
I'd say this is almost ready to be merged, except for:

* The semeval2012 analogy parameters should be changed to (0.05, 1.5).
* We shouldn't be running parameter tuning most of the time, just when running a particular evaluation for a paper or something. So I'd say the default on `tune_analogies` should be `False`.
* The docstring of `analogy.evaluate` should explain the parameters. (I'm a little unclear on what `semeval_scope` is.)

**Pull request merged**